agent/dns.go

```diff
@@ -1546,6 +1546,14 @@ func (d *DNSServer) serviceSRVRecords(cfg *dnsConfig, dc string, nodes structs.C
 					record.Hdr.Name = srvRec.Target
 					resp.Extra = append(resp.Extra, record)
 
+				// Alias to another Canonical Name
+				case *dns.CNAME:
+					if strings.HasSuffix(addr, ".") {
+						srvRec.Target = addr
+					} else {
+						srvRec.Target = fmt.Sprintf("%s.", addr)
+					}
```

Comment:
I think we still need to append any other records besides the CNAME one. In some cases Consul will recursively resolve this CNAME and have the A/AAAA record in the additional section.

So adding something like the following would be good. 

```
   if len(records) > 1 {
      resp.Extra = append(resp.Extra, records[1:]...)
   }
```